src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -126,24 +131,29 @@ export class ConfirmBid extends React.Component<ConfirmBidProps, ConformBidState
   }
 
   checkBidPosition(result) {
-    const bidderPosition = result.data.me.bidder_position
-    if (bidderPosition.processed_at) {
-      if (bidderPosition.is_active) {
-        // wining
-        this.showBidResult(true, "SUCCESS")
-      } else {
-        // outbid
-        this.showBidResult(false, "ERROR_BID_LOW")
-      }
-    } else {
+    const bidderPosition = result.data.me.bidder_position.position
+    const status = result.data.me.bidder_position.status
+    if (status === "SUCCESS") {
+      this.showBidResult(true, "SUCCESS")
+    } else if (status === "PENDING") {
       if (this.state.pollCount > MAX_POLL_ATTEMPTS) {
-        // TODO: Present error message to user.
+        const md = `We're receiving a high volume of traffic and your bid is still processing.  \
+If you don’t receive an update soon, please contact [support@artsy.net](mailto:support@artsy.net). `
+
+        this.showBidResult(false, "PROCESSING", "Bid Processing", md)
       } else {
         setTimeout(() => {
```

Comment:
It was implemented with `setInterval` but @ashfurrow and I changed it to `setTimeout` to make sure the response has been received before sending another request.

Reply:
Yeah exactly, so here's the scenario we need to be cautious of: a user is on a poor cell connection, lots of packet loss, there's gravity latency maybe, and if we trigger one new request every second, they might bunch up. The code Sepand and I worked on was changed so that a new request isn't sent until the previous request has returned a result, which is how Eidolon polling works. This means that polling twenty times at one-second intervals might actually take longer than twenty seconds, since we don't start a new request until the current one finishes, which is totally okay 👍 Maybe we should document that behaviour in a code comment?